osu.Game/Rulesets/RulesetStore.cs

```diff
@@ -48,7 +53,17 @@ public RulesetStore(IDatabaseContextFactory factory)
         /// </summary>
         public IEnumerable<RulesetInfo> AvailableRulesets { get; private set; }
 
-        private Assembly resolveRulesetAssembly(object sender, ResolveEventArgs args) => loadedAssemblies.Keys.FirstOrDefault(a => a.FullName == args.Name);
```

Comment:
it looks like this was here to cache assembly lookups. have you checked what happens (or even if it happens) when a ruleset assembly is resolved the second time?

Reply:
Ah, This makes sense now. 
However, after testing on master, this line doesn't have any effect effect because the event handler is attached after actual the assembly lookup is done (the event is never raised)
As of having a ruleset assembly being resolved twice; I don't have any scenario in mind where this could happen (apart from loading the same ruleset assembly but with different filenames, 
which is handled by the ruleset store)? Looks like I'm missing something.

Reply:
I'm not sure myself, but the first thing i'd do is git blame and see if there are any hints.

Reply:
After looking at the class commit log, that was added back in f2a9e95d.
Gonna add the loaded assembly lookup back as it seems to allow resolving assemblies without having full assembly info matching.